### PR TITLE
[Draft] Migrate GLM from GPTModel to HybridModel

### PR DESCRIPTION
**Pull request by xuantengh**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

# Changelog

- Add specific line by line info of high level changes in this PR.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci) in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: examples/model_verification_cards/glm5-2/card.yaml [hunk 1]
```diff
@@ -1,563 +1,251 @@
...
+    unverified:
+    - hf_to_megatron_cpu
+    - manual_forward_pass
   training:
     GB200:
-      verified: [pretrain, sft, sft_long_context, peft, checkpoint_resume]
-      unverified: [sft_export_inference]
-    H100:
-      verified: [pretrain, sft, sft_export_inference, sft_long_context, peft, checkpoint_resume]
+      unverified:
```
Comment (cuichenx) [inline examples/model_verification_cards/glm5-2/card.yaml:24]: Are you planning to verify these training paths as part of this PR or in a follow-up?

File: src/megatron/bridge/models/glm_moe_dsa/glm5_bridge.py [hunk 2]
```diff
@@ -72,26 +87,19 @@ def _should_map_hf_config_field(self, hf_config: Any, hf_name: str, megatron_nam
...
-        except (ImportError, ModuleNotFoundError):
-            logger.warning("DSA spec not available; falling back to standard GPT decoder block spec.")
-
         provider.normalization = "RMSNorm"
         provider.gated_linear_unit = True
         provider.add_bias_linear = False
         provider.share_embeddings_and_output_weights = False
         provider.qk_layernorm = True
         provider.multi_latent_attention = True
+        # Megatron Core DSA prohibits rope fusion
```
Comment (cuichenx) [inline src/megatron/bridge/models/glm_moe_dsa/glm5_bridge.py]: double check this claim?

File: src/megatron/bridge/models/glm_moe_dsa/glm5_hybrid.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        # so outstanding microbatches never reuse each other's top-k indices.
+        token = _FORWARD_STATE.set(({}, {}))
+        try:
+            return HybridStack.forward(stack, value, attention_mask, **kwargs)
+        finally:
+            _FORWARD_STATE.reset(token)
+
+    if stack.config.recompute_granularity == "full" and stack.training:
+        # Replay each sharing group together with fresh native DSA state.
+        def recompute(value: torch.Tensor) -> torch.Tensor:
```
Comment (cuichenx) [inline src/megatron/bridge/models/glm_moe_dsa/glm5_hybrid.py:88]: `value` could be None on non-first pipeline stages. HybridStack.forward retrieves self.input_tensor inside the closure. The checkpoint therefore receives no tensor requiring gradients, and its output loses the autograd graph.

**Comment by cuichenx** [comment]
Please also check the failed tests

File: examples/model_verification_cards/glm5-2/card.yaml (hunk 1 above)
Comment (Phlip79) [reply examples/model_verification_cards/glm5-2/card.yaml:24]: This is my fault due to my miscommunication. @xuantengh please verify items that were previously verified.
